api/views.py

```diff
@@ -281,6 +281,13 @@ def destroy(self, request, pk=None, **kwargs):
             if not tag.puzzles.exists():
                 tag.delete()
 
+            if puzzle.chat_room:
+                transaction.on_commit(
+                    lambda: puzzle.chat_room.get_service().handle_tag_removed(
```

Comment:
Hmm do you think `handle_tag_removed` makes more sense to be defined at the ChatRoom level?

I think initially, Akira and I agreed to not have the service exposed in the views (I know we kind of broke that rule with `announce()` but I kinda want to make that a static or class method under ChatRoom).

Let me know what you think?

Reply:
Sure, happy to move it to ChatRoom to make it more consistent with the existing code. I think the structure right now feels a little unintuitive to me because this isn't really specific to the particular chat room that's associated with the puzzle but rather with the Discord server as a whole, but we can worry about that kind of thing later.